geode-core/src/main/java/org/apache/geode/internal/cache/DiskStoreAttributes.java

```diff
@@ -112,6 +120,13 @@ public int getCompactionThreshold() {
     return result;
   }
 
+  public DiskDirSizesUnit getDiskDirSizesUnit() {
+    if (this.diskDirSizesUnit == null) {
+      this.diskDirSizesUnit = DEFAULT_DISK_DIR_SIZES_UNIT;
```

Comment:
I find it odd to be setting fields in a getter method. If the concern is that this field will be null upon deserialization, you could implement a `readObject` method that first calls `defaultReadObject` then checks the object's invariant by making sure `diskDirSizesUnit` is not null.

Reply:
I misunderstood one of the previous comments then, sorry. I checked how it is done in other classes, so I hope its fine now.